src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -348,9 +345,9 @@ public void EmitAndVerify(
             }
             catch (Exception)
             {
-                if (DumpAssembliesOnFailure)
+                if (DumpAssemblyLocation is string dumpPath)
```

Comment:
It looks like `Emit` already dumps whenever `DumpAssemblyLocation` is set. Why do we need to dump again when an exception is caught?